zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/DeploymentReconstructProcessorTest.java

```diff
@@ -487,4 +489,152 @@ private void assertHasFollowupCommand() {
+                .setFormId("form")
+                .setTenantId("form-tenant")
+                .setResourceName("form.form")
+                .setVersion(1));
+    state
+        .getDecisionState()
+        .storeDecisionRequirements(
+            new DecisionRequirementsRecord()
+                .setTenantId("decision-tenant")
+                .setDecisionRequirementsKey(decisionRequirementsKey));
+    state
+        .getDecisionState()
+        .storeDecisionRecord(
+            new DecisionRecord()
+                .setDeploymentKey(decisionDeploymentKey)
+                .setDecisionKey(decisionKey)
+                .setDecisionId("decision2")
+                .setDecisionName("decision2")
+                .setTenantId("decision-tenant")
+                .setDecisionRequirementsKey(decisionRequirementsKey)
+                .setVersion(1));
+    // when
+    processor.processRecord(command);
+
+    // then
+    assertThat(getCommandAt(eventIndex++).getValue())
+        .satisfies(
+            record -> {
+              assertThat(record.processesMetadata())
+                  .singleElement()
+                  .satisfies(el -> assertThat(el.getBpmnProcessId()).isEqualTo("process"));
+              assertThat(record.decisionsMetadata()).isEmpty();
+              assertThat(record.formMetadata()).isEmpty();
+            });
+
+    // when
+    processor.processRecord(getCommandAt(eventIndex++));
+
+    // then
+    assertThat(getCommandAt(eventIndex++).getValue())
+        .satisfies(
+            record -> {
+              assertThat(record.processesMetadata())
+                  .singleElement()
+                  .satisfies(el -> assertThat(el.getBpmnProcessId()).isEqualTo("process2"));
+              assertThat(record.decisionsMetadata()).isEmpty();
+              assertThat(record.formMetadata()).isEmpty();
+            });
+
+    // when
+    processor.processRecord(getCommandAt(eventIndex++));
+
+    // then
+    assertThat(getCommandAt(eventIndex++).getValue())
+        .satisfies(
+            record -> {
+              assertThat(record.processesMetadata()).isEmpty();
+              assertThat(record.decisionsMetadata()).isEmpty();
+              assertThat(record.formMetadata())
+                  .singleElement()
+                  .satisfies(el -> assertThat(el.getFormKey()).isEqualTo(formKey));
+            });
+
+    // when
+    processor.processRecord(getCommandAt(eventIndex++));
+
+    // then
+    assertThat(getCommandAt(eventIndex++).getValue())
+        .satisfies(
+            record -> {
+              assertThat(record.processesMetadata()).isEmpty();
+              assertThat(record.formMetadata()).isEmpty();
+              assertThat(record.decisionsMetadata())
+                  .singleElement()
+                  .satisfies(el -> assertThat(el.getDecisionKey()).isEqualTo(decisionKey));
+            });
+
+    // when
+    // let's run again the processor to make it reach progressState = Done
```

Comment:
:wrench: Let's assert that

Reply:
Unfortunately the progressState is not saved anymore in the class. ProgressState.DONE is never written inside a command so I dont think we can assert that in any way, we just need to check that we reach "RECONSTRUCTED_ALL" as an intent.

Reply:
That's fine too :+1: